canonical-web-and-design/maas-ui
# fix(ui): add missing edit bcache/RAID forms using generic EditDisk component

## Done

- Refactored `EditPhysicalDisk` into a generic `EditDisk` component for use with all disk types (except volume groups)
- Removed `EditLogicalVolume`

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Change the storage layout to blank
- Check that you can edit the physical disk from the action menu
- Partition the disk and use it to create a cache set
- Make another partition and create a bcache from it
- Check that you can edit the bcache from the action menu
- Remove the cache set and bcache
- Create a volume group from the two partitions
- Check that the volume group action menu does not include an edit option
- Add a logical volume from the volume group
- Check that you can edit the logical volume from the action menu
- Remove the logical volume and volume group
- Create a RAID from the two partitions
- Check that you can edit the RAID device from the action menu

## Fixes

Fixes #2116
